### PR TITLE
sam do: improved the help

### PR DESCRIPTION
**Pull request by cchampet**

### CODE REVIEW

File: applications/sam/sam_do.py [hunk 1]
```diff
@@ -31,80 +31,58 @@ def __init__(self):
...
 
     ''' + self._imgSeqConversion + '''
         Convert Image:                     sam do reader in.dpx // writer out.jpg
         Convert Sequence:                  sam do reader in.####.dpx // writer out.####.jpg
+        Select a frame:                    sam do reader in.####.dpx // writer out.####.jpg --ranges 10
         Select a range:                    sam do reader in.####.dpx // writer out.####.jpg --ranges 10 100
         Select several ranges:             sam do reader in.####.dpx // writer out.####.jpg --ranges 10 100 150 200
-        Note:                              'r' and 'w' are shortcuts for 'reader' and 'writer'
+        Convert files in directory:        sam do reader inDir // writer outDir ext=jpg
+        Convert some files in directory:   sam do reader inDir ext=dpx // writer outDir ext=jpg
```
Comment (fabiencastan) [inline applications/sam/sam_do.py]:
```
Convert files in directory recursively:        sam do reader inDir // writer outDir ext=jpg
Convert some files in directory:   sam do reader inDir ext=dpx // writer outDir ext=jpg --no-recursivity
```

File: applications/sam/sam_do.py [hunk 2]
```diff
@@ -31,80 +31,58 @@ def __init__(self):
...
-        Checkerboard generator:            sam do checkerboard width=500  // viewer
-        Checkerboard generator:            sam do checkerboard width=1920 ratio=2.35 // viewer
-        Text writing:                      sam do constant // text text="hello" size=80 // viewer
+    ''' + self._principle + '''
+        'sam do' can be seen as a command line version of a compositing software in which you can chain nodes, with a directed acyclic graph.
+        The general syntax is:             sam do plugin // plugin // plugin
+        Which plugins are available?       sam do --nodes
+        How does a plugin work?            sam do blur -h
 
     ''' + self._imgSeqConversion + '''
```
Comment (fabiencastan) [inline applications/sam/sam_do.py:56]: Could be useful to have an example with `--continue-on-error`

File: applications/sam/sam_do.py (hunk 1 above)
Comment (cchampet) [inline applications/sam/sam_do.py]: a034f35

File: applications/sam/sam_do.py (hunk 2 above)
Comment (cchampet) [inline applications/sam/sam_do.py:56]: a3f8b3c

**Pull request merged**